### DISCUSSION TITLE
How to setup a reverse proxy?

### DISCUSSION THREAD

**Discussion by brunolm** [vercel/community#1832] [Help] [answered]
I want to setup Ghost in a subdirectory of my website, however the instructions only include `Nginx`, `Apache`, `Cloudfront`.
https://ghost.org/help/can-i-run-ghost-from-a-subdirectory/

I'm wondering if Vercel allows doing something equivalent, or if it allows us to setup Nginx somehow. Is it possible?

**Comment by Nick-Mazuk** [comment]
You may wish to look at Vercel's rewrites: https://vercel.com/docs/configuration#project/rewrites

With the rewrites, you can point example.com/ghost to your ghost site, and it will mask the domain so it looks like it's still coming from your original domain.

**Comment by quyctd** [comment]
@brunolm Hi, I have the same use case but after `rewrites`, I still cannot access the ghost blog and got redirected to https://error.ghost.org/. Do you know what wrong? Here is my `rewrites` function:

```js
async rewrites() {
        return [
            {
                source: '/blog/:path*',
                destination: 'https://example.ghost.io/:path*',
            },
        ];
    },
```

**Comment by brunolm** [comment, marked as answer]
I end up with this in `vercel.json` for the subfolder `/blog` (note that trailing `/` **IS REQUIRED**).

You also need to contact them for the domain change, you can't do it yourself.

```json
{
  "routes": [
    {
      "src": "/blog",
      "dest": "https://YOURGHOST.ghost.io/blog/"
    },

    {
      "src": "/blog/(.*?)([.]\\w{2,5})(\\?[^/]+)?",
      "dest": "https://YOURGHOST.ghost.io/blog/$1$2$3"
    },

    {
      "src": "/blog/([\\S\\s]+)",
      "dest": "https://YOURGHOST.ghost.io/blog/$1/"
    }
  ]
}
```

**Reply by quyctd** [reply]
Thank you very much 👍

**Reply by jakobgreenfeld** [reply]
Who do you need to contact for which domain change?

**Reply by holden** [reply]
This didn't quite work for me, which might have to do with self-hosting or changes within the last 2 years, but this worked:

```
{
  "routes": [
    {
      "src": "/blog/?", # for both /blog and /blog/
      "dest": "https://blog.YOURHOST.com/blog/"
    },
    {
      "src": "/blog/(.*?)([.]\\w{2,5})(\\?[^/]+)?",
      "dest": "https://blog.YOURHOST.com/blog/$1$2$3"
    },
    {
      "src": "/blog/([\\S\\s]+?)/?", # had to add extra option as it was not working properly
      "dest": "https://blog.YOURHOST.com/blog/$1/"
    }
  ]
}
```

**Comment by ugokoli** [comment]
In addition to @brunolm comment, this is what works for me in next.config.js
Reference doc https://ghost.org/help/can-i-run-ghost-from-a-subdirectory/

```

  async rewrites() {
    return [
      {
        source: "/blog",
        destination: "https://YOURGHOST.ghost.io/blog/",
      },
      {
        source: "/blog/(.*?)([.]\\w{2,5})(\\?[^/]+)?",
        destination: "https://YOURGHOST.ghost.io/blog/$1$2$3",
      },
      {
        source: "/blog/([\\S\\s]+)",
        destination: "https://YOURGHOST.ghost.io/blog/$1/",
      },
    ];
  },
  async headers() {
    return [
      {
        source: "/blog/:path*",
        headers: [
          { key: "X-Forwarded-Proto", value: "https" },
          { key: "X-Forwarded-Host", value: YOURDOMAIN },
          // { key: "X-Forwarded-For", value: "" },
        ],
      },
    ];
  },
```

**Reply by elie222** [reply]
what's the purpose of the headers?
